### DISCUSSION TITLE
ESP32-CAM stream is not covered by WebUI password protection

### DISCUSSION THREAD

**Discussion by samazw** [arendst/Tasmota#12983] [Ideas]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
The Tasmota version for ESP32-Camera module has the option "set password".
But this password protection does not cover the direct link to the video stream.

Meaning: The password protect the webinterface of tasmota itself but not http://DEVICE_IP:81/stream.
So correct would be the need of user+password also for the stream.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_
- enable password protection on Tasmota ESP32CAM Module
- open a browser window and insert direct link to camera stream, eg http://192.168.10.10:81/stream
- Result: Stream can be opened and is shown without a password.

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
An user + password should be required to get the stream.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

This is not an ENHANCEMENT (as labeled by ascillato2 but an SECURITY issue. Stream is Not protected.

**Comment by martink1979** [comment]
Gibt es schon eine Lösung für dieses Problem ?  Das die Stream URL nicht mit einem Passwort geschützt ist ? 

GOOGLE TRANSLATE: Is there already a solution for this problem? That the stream URL is not protected with a password?

**Comment by sfromis** [comment]
It is not a "problem", merely a wish for an enhancement.

**Comment by martink1979** [comment]
> It is not a "problem", merely a wish for an enhancement.

Gibt es denn schon einen workaround dazu ? Das man den stream irgendwie schützen kann ? 

GOOGLE TRANSLATE: Is there already a workaround for this? That you can somehow protect the stream?

**Comment by ascillato2** [comment]
> GOOGLE TRANSLATE: Is there already a workaround for this? That you can somehow protect the stream?

Not at this moment, that is why this issue is open to anyone that want to jump in and make a PR.

**Comment by Jason2866** [comment]
Moving to discussions since it is not a issue

**Comment by Nemobi** [comment]
I've been looking into this and I can make a PR for adding basic auth to the mjpeg stream.
I added a new command, `WcAuth` for enabling and disabling authentication on the stream separate from the WebUI. When turned on it uses the WebUI password for authentication.

**Reply by ascillato** [reply]
Thank you very much. 👍 

Just for reference, @Nemobi's PR is https://github.com/arendst/Tasmota/pull/15687

**Reply by Jason2866** [reply]
Merged and available in latest development.

**Reply by samazw** [reply]
Thanks a lot colleagues - this is great!
